examples/kubernetes-deployment-example.yaml

```diff
@@ -0,0 +1,166 @@
+        workingDir: /opt/almond
+      - name: nginx-sidecar
+        image: nginx:alpine
+        env:
+        - name: TZ
+          value: Europe/London
+        volumeMounts:
+        - name: nginx-sidecar-config
+          mountPath: /etc/nginx/nginx.conf
+          subPath: nginx.conf
+      volumes:
+      - hostPath:
+          path: /run/user/1000/pulse
+          type: Directory
+        name: run-user-1000
+      - hostPath:
+          path: /home/pi/.config/almond-server
+          type: Directory
+        name: home-user-config-almond-server
+      - name: nginx-sidecar-config
+        configMap:
+          name: nginx-sidecar-conf
+status: {}
+
+---
+
+# SERVICE
+# Uses nodeport for local access
+
+apiVersion: v1
+kind: Service
+metadata:
+  name: almond-service
+  # namespace <your namespace>
+  labels:
+    app: almond-service
+spec:
+  type: NodePort
+  ports:
+  - name: almond-web
+    port: 80
+    targetPort: 3001
+    protocol: TCP
+    nodePort: 32000
+  selector:
+    app: almond-server
+    role: backend
+status: {}
+
+---
+
+# CONFIGMAP
+# nginx sidecar configmap
+# modify the allow section to allow only host you trust; deny all otherwise
+# monitor nginx logs for errors: e.g. kubectl logs -n namespace -c nginx-sidecar <pod id>
+
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: nginx-sidecar-conf
+  # namespace: <your namespace>
+data:
+  nginx.conf: |-
+    user nginx;
+    worker_processes  1;
+    pid /var/run/nginx.pid;
+    error_log /dev/stdout info;
+
+    events {
+        worker_connections 1024;
+    }
+
+    http {
+        include             mime.types;
+        default_type        application/octet-stream;
+        sendfile            on;
+        keepalive_timeout   65;
+        proxy_read_timeout  1d;
+        gzip                on;
+        gzip_disable        "msie6";
+
+        upstream backend {
+            ip_hash;
+            server 127.0.0.1:3000;
+            #server almond-service:3000;
+        }
+      
+        map $http_upgrade $connection_upgrade {
+            default upgrade;
+            ''      close;
+        }
+      
+        # API & Ingress
+        server {
+            listen 3001 default_server;
+            listen 8099 default_server;
+            listen [::]:3001 default_server;
+            listen [::]:8099 default_server;
+      
+            allow   127.0.0.1;  # localhost
+            deny    all;
+      
+            server_name _;
+            access_log /dev/stdout combined;
+      
+            client_max_body_size 4G;
+            keepalive_timeout 5;
+      
+            root /dev/null;
+      
+            location / {
+                proxy_redirect off;
+                proxy_pass http://backend;
+      
+                proxy_http_version 1.1;
+                proxy_set_header Upgrade $http_upgrade;
+                proxy_set_header Connection $connection_upgrade;
+                proxy_set_header Host "127.0.0.1:3000";
```

Comment:
Setting this might confuse the logic to compute the right redirect URLs for OAuth. DId you try with any OAuth skill (such as Spotify)?

Reply:
Same as previous comment; will investigate and report back.

Reply:
Indeed setting the header in that way dose confuse the logic because when you try to configure Spotify for example although you are presented with your Spotify account login page, the end result is: ‘ cannot open page because it cannot connet to the server’. More ideas needed.
But isn’t this a restriction on the local almond server or have there been some other changes since 1.8.0?

Reply:
There has been changes here as we try to make the experience smoother for users of the Home Assistant add-on. I additionally changed how we check for host-based authentication and how the origin check works, in a way that should make this hack less necessary.